ksqldb-functional-tests/src/test/resources/query-validation-tests/partition-by.json

```diff
@@ -1025,6 +1025,140 @@
         "type": "io.confluent.ksql.util.KsqlStatementException",
         "message": "Map keys, including types that contain maps, are not supported as they may lead to unexpected behavior due to inconsistent serialization. Key column name: `VAL`. Column type: STRUCT<`F1` MAP<STRING, INTEGER>>. See https://github.com/confluentinc/ksql/issues/6621 for more."
       }
+    },
+    {
+      "name": "multiple columns - select star",
+      "statements": [
+        "CREATE STREAM INPUT (NAME STRING, ID INT, AGE INT) with (kafka_topic='input', format='JSON');",
+        "CREATE STREAM OUTPUT AS select * from INPUT partition by ID, AGE;"
+      ],
+      "inputs": [{"topic": "input", "value": {"NAME": "bob", "ID": 10, "AGE": 30}}],
+      "outputs": [{"topic": "OUTPUT", "key": {"ID": 10, "AGE": 30}, "value": {"NAME": "bob"}}],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, AGE INT KEY, NAME STRING"}
+        ]
+      }
+    },
+    {
+      "name": "multiple columns - select explicit",
+      "statements": [
+        "CREATE STREAM INPUT (NAME STRING, ID INT, AGE INT) with (kafka_topic='input', format='JSON');",
+        "CREATE STREAM OUTPUT AS select ID, AGE, NAME from INPUT partition by ID, AGE;"
+      ],
+      "inputs": [{"topic": "input", "value": {"NAME": "bob", "ID": 10, "AGE": 30}}],
+      "outputs": [{"topic": "OUTPUT", "key": {"ID": 10, "AGE": 30}, "value": {"NAME": "bob"}}],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, AGE INT KEY, NAME STRING"}
+        ]
+      }
+    },
+    {
+      "name": "multiple columns - some key some value",
+      "statements": [
+        "CREATE STREAM INPUT (NAME STRING KEY, ID INT, AGE INT) with (kafka_topic='input', format='JSON');",
+        "CREATE STREAM OUTPUT AS select ID, AGE, NAME from INPUT partition by ID, NAME;"
+      ],
+      "inputs": [{"topic": "input", "key": "bob", "value": {"ID": 10, "AGE": 30}}],
+      "outputs": [{"topic": "OUTPUT", "key": {"ID": 10, "NAME": "bob"}, "value": {"AGE": 30}}],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, NAME STRING KEY, AGE INT"}
+        ]
+      }
+    },
+    {
+      "name": "multiple key columns",
+      "statements": [
+        "CREATE STREAM INPUT (NAME STRING KEY, ID INT KEY, AGE INT) with (kafka_topic='input', format='JSON');",
+        "CREATE STREAM OUTPUT AS select ID, AGE, NAME from INPUT partition by ID, NAME;"
+      ],
+      "inputs": [{"topic": "input", "key": {"NAME": "bob", "ID": 10}, "value": {"AGE": 30}}],
+      "outputs": [{"topic": "OUTPUT", "key": {"ID": 10, "NAME": "bob"}, "value": {"AGE": 30}}],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, NAME STRING KEY, AGE INT"}
+        ]
+      }
+    },
+    {
+      "name": "multiple columns with expressions",
```

Comment:
can we name one of the expressions in the projection to make sure that the key name is properly propagated to the schema?